### PR TITLE
[8.0] web_action_conditionable

### PR DESCRIPTION
**Pull request by ovnicraft**
This module implements condition for actions in tree view.
do allows: edit=“false” create=“false” delete=“false”
This module allows too: delete=“state==‘draft’”, simple expression.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Add a README according OCA's template (https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst) explaining the usage of the module.

File: web_action_conditionable/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "name": 'web_action_conditionable',
+    "version": "0.1",
+    "depends": [
+        'base',
+        'web',
+    ],
+    'data': ['views/view.xml'],
+    "author": "Cristian Salamea,Odoo Community Association (OCA)",
+    "installable": True,
+    "active": False,
```
Comment (dreispt) [inline web_action_conditionable/__openerp__.py]: The `active`key is deprecated.

**Comment by dreispt** [comment]
Needs the README with the module description.

**Comment by ovnicraft** [comment]
@pedrobaeza @dreispt done.

File: web_action_conditionable/README.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Web Action Conditionable
+========================
+
+This module was written to extend the functionality of actions in tree views.
+Odoo by default support delete="false" create="false", with this module you can:
+delete="state=='draft'".
```
Comment (dreispt) [inline web_action_conditionable/README.rst]: Use backticks around the code samples.

File: web_action_conditionable/README.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Installation
+============
+
+To install this module, just follow basic steps to install an odoo module.
+
+
+Configuration
+=============
+
+No needed.
```
Comment (dreispt) [inline web_action_conditionable/README.rst]: s/No/Not

**Comment by dreispt** [comment]
LGTM (no test) :+1:

**Comment by ovnicraft** [comment]
@dreispt ok looks all is green :+1:

File: web_action_conditionable/static/src/js/views.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*global openerp, _, $ */
+
+openerp.web_action_conditionable = function (instance) {
+  instance.web.View.include({
+    is_action_enabled: function(action) {
+      var attrs = this.fields_view.arch.attrs;
+      if (action in attrs) {
+	try {
+	  data = JSON.parse(attrs[action]);
```
Comment (hbrunn) [inline web_action_conditionable/static/src/js/views.js]: don't do that yourself, call super here

File: web_action_conditionable/static/src/js/views.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	  return data;
+	} catch(error) {
```
Comment (hbrunn) [inline web_action_conditionable/static/src/js/views.js:10]: does it make sense to inspect error and only do this if we had a json parse error?

File: web_action_conditionable/static/src/js/views.js (hunk 4 above)
Comment (ovnicraft) [inline web_action_conditionable/static/src/js/views.js]: why not ? call super gives any better point here ?

File: web_action_conditionable/static/src/js/views.js (hunk 5 above)
Comment (ovnicraft) [inline web_action_conditionable/static/src/js/views.js:10]: @hbrunn attrs comes with "true" when developer puts explicit and this module wants to support simple expression to decide.
parse  `state=='draft'` gives syntax error

File: web_action_conditionable/static/src/js/views.js (hunk 4 above)
Comment (hbrunn) [inline web_action_conditionable/static/src/js/views.js]: you don't know what happens in the future, and also not what other modules exist that might also do whatever with this function. By calling super and only acting on failures, you give other code the chance to do whatever that code is good for

File: web_action_conditionable/static/src/js/views.js (hunk 5 above)
Comment (hbrunn) [inline web_action_conditionable/static/src/js/views.js:10]: great, then you can do your parsing when super raises a syntax error

Comment (ovnicraft) [inline web_action_conditionable/static/src/js/views.js:10]: will try your suggestion

**Comment by ovnicraft** [comment]
@hbrunn sorry for delay, its fixed.

**Comment by hbrunn** [comment]
@ovnicraft no worries :+1:

**Pull request merged**